python/sglang/srt/model_executor/model_runner.py

```diff
@@ -245,8 +247,10 @@ def load_model(self):
             load_format=self.server_args.load_format,
             download_dir=self.server_args.download_dir,
         )
+        monkey_patch_vllm_model_config(self.is_generation)
         self.vllm_model_config = VllmModelConfig(
             model=self.server_args.model_path,
+            task="generate" if self.model_config.is_generation else "embedding",
```

Comment:
Can we try to always return "generate" here and always return "generate" in `_resolve_task`?

Reply:
The change is meaningless, we don't use this parameter in the patched `_resolve_task`. It's only for code readability.

Reply:
Then can we just drop this argument? Because later vllm might introduce more and we do not want to follow that.
Also, what if we just return a dummy value in `_resolve_task`? What are the error messages?

Reply:
This is a breaking change and the `task` field is a **required** field, that's why I add it here.

Reply:
You can understand it this way: the `task` field is added to ModelConfig because it is required when initializing ModelConfig, it no longer has anything to do with _resolve_task, and we currently use a dummy value in `_resolve_task`. I write generate or embedding here just for code readability, to reflect the semantics of this field.